app/layout.tsx

```diff
@@ -23,10 +24,12 @@ export default function RootLayout({
   return (
     <html lang='en'>
       <body
-        className={`${poppins.variable} min-h-screen bg-white-800 px-2 font-poppins dark:bg-black-300 sm:px-12`}
+        className={`${poppins.variable} min-h-screen bg-white-800 px-4 font-poppins dark:bg-black-300 sm:px-12 md:px-24`}
       >
-        <Navbar />
-        {children}
+        <ThemeProvider>
```

Comment:
```suggestion
        <ThemeProvider defaultTheme={theme?.value || ''}>
```
Then we provide the theme from our cookies to our provider so they stay in sync. 

Basically whatever theme we set in our cookies, we tell our theme provider in the app to start with. This prevents the flicker of the theme on a refresh.